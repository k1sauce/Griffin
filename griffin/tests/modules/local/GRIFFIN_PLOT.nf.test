nextflow_process {

    name "Test Process GRIFFIN_PLOT"
    script "modules/local/GRIFFIN_PLOT.nf"
    process "GRIFFIN_PLOT"
    tag "GRIFFIN_PLOT"

    //
    test("GRIFFIN_PLOT_T01") {
        when {
            process {
                /*
                tuple val(meta), path(uncorrected_coverage_tsv), path(gc_corrected_coverage_tsv)
                val(save_window)
                val(step)
                */
                """
                input[0] = [
                    [id: 'Healthy_demo', bam_file_name: 'Healthy_demo'],
                    file("${projectDir}/test_data/griffin_griffin_nucleosome_profiling_griffin_merge_sites/Healthy_demo.uncorrected.coverage.tsv"),
                    file("${projectDir}/test_data/griffin_griffin_nucleosome_profiling_griffin_merge_sites/Healthy_demo.GC_corrected.coverage.tsv")
                ]
                input[1] = [-1000, 1000]
                input[2] = 15
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.result).match("GRIFFIN_PLOT_T01")
        }
    }

}