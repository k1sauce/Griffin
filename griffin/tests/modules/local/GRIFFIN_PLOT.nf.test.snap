{
    "GRIFFIN_PLOT_T01": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-28T21:24:33.048508"
    }
}